### PR TITLE
[KAT-1848] logging: add no-op tracer for benchmarking and use 64bit integers for…

### PR DESCRIPTION
**Pull request by patrickkenney9801**
… ids in json tracer

Add a no-op tracer for benchmarking
Use 64-bit integers for ids for better Jaeger integration and faster generation

### CODE REVIEW

**Comment by srogatch** [comment]
What is the motivation for having a separate tracing rather than using `katana/Logging.h` , perhaps extended with `KATANA_LOG_TRACE()` macro? The macro could be a real no-op without tracing, rather than the instantiation of smart pointers and other heap allocations.

**Comment by patrickkenney9801** [comment]
There is more information in tracing than simply printing a string so it cannot be made into a macro
We've also been planning on adding a no-op specific tracer for a while so we can benchmark overhead
If we decide to use Logging.h's methods later we can do so, but for now this looks like it solves both of our issues

File: libsupport/src/JSONTracer.cpp [hunk 1]
```diff
@@ -4,22 +4,23 @@
 
 #include <cstring>
 #include <iostream>
+#include <limits>
 #include <mutex>
 
 #include <arrow/memory_pool.h>
+#include <katana/Random.h>
```
Comment (tylershunt) [inline libsupport/src/JSONTracer.cpp]: Revert this change please. Should be `"katana/Random.h"` since that's our convention for headers that we own.

**Review by tylershunt — APPROVED** [review]
LGTM
